conan/internal/cache/db/packages_table.py

```diff
@@ -198,3 +215,28 @@ def get_package_references(self, ref: RecipeReference, only_latest_prev=True):
             r = conn.execute(query)
             for row in r.fetchall():
                 yield self._as_dict(self.row_type(*row))
+
+    def get_package_references_with_build_id_match(self, ref: RecipeReference, build_id):
+        # Return the latest revisions
+        assert ref.revision, "To search for package id's you must provide a recipe revision."
+        # we select the latest prev for each package_id
+        query = f'SELECT {self.columns.reference}, ' \
```

Comment:
This is repeated from above, which sucks, should probably refactor both out? But I like the explicitness